gammapy/astro/darkmatter/spectra.py

```diff
@@ -3,26 +3,37 @@
 import numpy as np
 import astropy.units as u
 from astropy.table import Table
+from gammapy.maps import MapAxis, RegionNDMap
 from gammapy.modeling import Parameter
 from gammapy.modeling.models import (
     SPECTRAL_MODEL_REGISTRY,
     SpectralModel,
+    TemplateNDSpectralModel,
     TemplateSpectralModel,
 )
+from gammapy.utils.deprecation import deprecated
 from gammapy.utils.interpolation import LogScale
 from gammapy.utils.scripts import make_path
 
 __all__ = ["PrimaryFlux", "DarkMatterAnnihilationSpectralModel"]
 
 
-class PrimaryFlux:
+class PrimaryFlux(TemplateNDSpectralModel):
     """DM-annihilation gamma-ray spectra.
 
     Based on the precomputed models by Cirelli et al. (2016). All available
     annihilation channels can be found there. The dark matter mass will be set
     to the nearest available value. The spectra will be available as
-    `~gammapy.modeling.models.TemplateSpectralModel` for a chosen dark matter mass and
-    annihilation channel.
+    `~gammapy.modeling.models.TemplateNDSpectralModel` for a chosen dark matter mass and
+    annihilation channel. Using a `~gammapy.modeling.models.TemplateNDSpectralModel`
+    allows the interpolation beween different dark matter masses.
+
+    Parameters
+    ----------
+    mDM : `~astropy.units.Quantity`
+        Dark matter mass
```

Comment:
While this may be obvious, it could be worth to mention that the mass is given as rest mass energy (ie m c^2)